php_phongo.c

```diff
@@ -1561,13 +1561,13 @@ static void php_phongo_cursor_iterator_dtor(zend_object_iterator *iter TSRMLS_DC
 {
 	php_phongo_cursor_iterator *cursor_it = (php_phongo_cursor_iterator *)iter;
 
+	php_phongo_cursor_free_current(cursor_it->cursor);
+
```

Comment:
cursor_it->cursor is the php_phongo_cursor_t* of the cursor_it->intern.data zval*.

If you destroy the zval first, the php_phongo_cursor_t* will be invalid.